p2p/subscriber_metrics.go

```diff
@@ -0,0 +1,130 @@
+package p2p
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const (
+	statusKey    = "status"
+	statusAccept = "accept"
+	statusIgnore = "ignore"
+	statusReject = "reject"
+)
+
+type subscriberMetrics struct {
+	messageNumInst  metric.Int64Counter
+	messageSizeInst metric.Int64Histogram
+
+	messageTimeLastMu sync.Mutex
+	messageTimeLast   time.Time
+	messageTimeInst   metric.Float64Histogram
+
+	subscriptionNum     atomic.Int64
+	subscriptionNumInst metric.Int64ObservableGauge
+	subscriptionNumReg  metric.Registration
```

Comment:
do we wanna store the Registration or just the unsub func ?

Reply:
I don't see any difference

Reply:
doesn't matter, more of a nit as to what we actually want to do with metricRegistration since we only need it for that unsub func